File: Dataset/Leetcode/train/112/164.java
class Solution {
    public boolean XXX(TreeNode root, int targetSum) {
        int sum = 0;
        if (root == null) return false;
        return dfs(root, targetSum, sum);
    }
    boolean dfs(TreeNode root, int target, int sum){
        if (root == null) {
            return false;
         }
         sum += root.val;
        if (root.left == null && root.right == null) {
            return sum == target;
        }
       return  dfs(root.left, target, sum) || dfs(root.right, target, sum);
    }
}

File: Dataset/Leetcode/valid/101/164.cpp
 class Solution {
public:
    bool XXX(TreeNode* root) {
        if(!root)
            return true;
        return isEqual(root->left, root->right);
    }
    bool isEqual(TreeNode* l,TreeNode* r)
    {
        if(l==NULL||r==NULL)
        {
            if(l==NULL&&right==NULL)
            {
                return true;
            }
            else
                return false;
        }
        if(l->val==r->val)
        {
            return isEqual(l->left,r->right) && isEqual(l->right,r->left);
        }
        return false;
    }
};

